src/macros.rs

```diff
@@ -150,3 +150,25 @@ macro_rules! wrap_pymodule {
         }
     };
 }
+
+#[cfg(not(PyPy))]
+/// Add current module to the existing table of built-in modules.
+///
+/// Use it before [`prepare_freethreaded_python`](crate::prepare_freethreaded_python) and
+/// leave feature `auto-initialize` off
```

Comment:
To be honest, I wasn't aware of this function, and I think the vast majority of people reading this documentation are not either.

In particular, a reader might have the following questions or thoughts:
- I want to add a module, this is what I need (is it?)
- What does this do and why should I use it?
- What is the point of adding builtin modules?
- You said you use this to debug modules, can you explain? How can a user use this for debugging?

Reply:
Added here and `building_and_distribution.md`